File: port_js/org/kbinani/vsq/NoteHeadHandle.js
/*
 * NoteHeadHandle.js
 * Copyright (C) 2009-2010 kbinani
 *
 * This file is part of org.kbinani.vsq.
 *
 * org.kbinani.vsq is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * org.kbinani.vsq is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
if( org == undefined ) var org = {};
if( org.kbinani == undefined ) org.kbinani = {};
if( org.kbinani.vsq == undefined ) org.kbinani.vsq = {};
if( org.kbinani.vsq.NoteHeadHandle == undefined ){

    org.kbinani.vsq.NoteHeadHandle = function(){
        // copy of IconParameter.ctor =>
        /// <summary>
        /// アイコン設定の種類
        /// [ArticulationType]
        /// </summary>
        this.articulation = org.kbinani.vsq.ArticulationType.Dynaff;
        /// <summary>
        /// アイコンのボタンに使用される画像ファイルへの相対パス
        /// </summary>
        this.button = "";
        /// <summary>
        /// キャプション
        /// </summary>
        this.caption = "";

        /// <summary>
        /// ゲートタイム長さ
        /// </summary>
        this.length = 0;
        /// <summary>
        /// ビブラート深さの開始値
        /// </summary>
        this.startDepth = 64;
        /// <summary>
        /// ビブラート深さの終了値
        /// </summary>
        this.endDepth = 64;
        /// <summary>
        /// ビブラート速さの開始値
        /// </summary>
        this.startRate = 64;
        /// <summary>
        /// ビブラート速さの終了値
        /// </summary>
        this.endRate = 64;
        this.startDyn = 64;
        this.endDyn = 64;
        this.duration = 1;
        this.depth = 64;
        this.dynBP = null;
        this.depthBP = null;
        this.rateBP = null;
        this.buttonImageFullPath = "";
        // <=

        this.articulation = org.kbinani.vsq.ArticulationType.NoteAttack;
        this.Index = 0;
        this.IconID = "";
        this.IDS = "";
        this.Original = 0;
        if( arguments.length >= 3 ){
            this.IDS = arguments[0];
            this.IconID = arguments[1];
            this.Index = arguments[2];
        }
    };

    org.kbinani.vsq.NoteHeadHandle.prototype = new org.kbinani.vsq.IconParameter();

    /**
     * @return [String]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.toString = function() {
        return getDisplayString();
    };

    /**
     * @return [int]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.getDepth = function() {
        return this.depth;
    };

    /**
     * @param value [int]
     * @return [void]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.setDepth = function( value ) {
        this.depth = value;
    };

    /**
     * @return [int]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.getDuration = function() {
        return this.duration;
    };

    /**
     * @param value [int]
     * @return [void]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.setDuration = function( value ) {
        this.duration = value;
    };

    /**
     * @return [String]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.getCaption = function() {
        return this.caption;
    };

    /**
     * @param value [String]
     * @return [void]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.setCaption = function( value ) {
        this.caption = value;
    };

    /**
     * @return [int]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.getLength = function() {
        return this.length;
    };

    /**
     * @param value [int]
     * @return [void]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.setLength = function( value ) {
        this.length = value;
    };

    /**
     * @return [String]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.getDisplayString = function() {
        return this.IDS + this.caption;
    };

    /**
     * @return [object]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.clone = function() {
        var result = new org.kbinani.vsq.NoteHeadHandle();
        result.Index = this.Index;
        result.IconID = this.IconID;
        result.IDS = this.IDS;
        result.Original = this.Original;
        result.setCaption( getCaption() );
        result.setLength( getLength() );
        result.setDuration( getDuration() );
        result.setDepth( getDepth() );
        return result;
    };

    /**
     * @return [VsqHandle]
     */
    org.kbinani.vsq.NoteHeadHandle.prototype.castToVsqHandle = function() {
        var ret = new VsqHandle();
        ret.m_type = org.kbinani.vsq.VsqHandleType.NoteHeadHandle;
        ret.Index = this.Index;
        ret.IconID = this.IconID;
        ret.IDS = this.IDS;
        ret.Original = this.Original;
        ret.Caption = getCaption();
        ret.setLength( getLength() );
        ret.Duration = getDuration();
        ret.Depth = getDepth();
        return ret;
    };

}

File: org.kbinani.vsq/NoteHeadHandle.cs
/*
 * NoteHeadHandle.cs
 * Copyright © 2009-2011 kbinani
 *
 * This file is part of org.kbinani.vsq.
 *
 * org.kbinani.vsq is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * org.kbinani.vsq is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
#if JAVA
package org.kbinani.vsq;

import java.io.*;
#else
using System;

namespace org.kbinani.vsq {
#endif

#if JAVA
    public class NoteHeadHandle extends IconParameter implements Cloneable, Serializable {
#else
    [Serializable]
    public class NoteHeadHandle : IconParameter, ICloneable {
#endif
        public int Index;
        public String IconID = "";
        public String IDS = "";
        public int Original;

        public NoteHeadHandle() {
        }

        public NoteHeadHandle( String aic_file, String ids, String icon_id, int index )
#if JAVA
        {
#else
            :
#endif
            base( aic_file )
#if JAVA
            ;
#else
        {
#endif
            IDS = ids;
            IconID = icon_id;
            Index = index;
        }

        public String toString() {
            return getDisplayString();
        }

#if !JAVA
        public override String ToString(){
            return toString();
        }
#endif

#if !JAVA
        /// <summary>
        /// XMLシリアライズ用
        /// </summary>
        public int Depth{
            get{
                return getDepth();
            }
            set{
                setDepth( value );
            }
        }
#endif

        public int getDepth() {
            return depth;
        }

        public void setDepth( int value ) {
            depth = value;
        }

#if !JAVA
        /// <summary>
        /// XMLシリアライズ用
        /// </summary>
        public int Duration {
            get {
                return getDuration();
            }
            set {
                setDuration( value );
            }
        }
#endif

        public int getDuration() {
            return duration;
        }

        public void setDuration( int value ) {
            duration = value;
        }

#if !JAVA
        /// <summary>
        /// XMLシリアライズ用
        /// </summary>
        public String Caption {
            get {
                return getCaption();
            }
            set {
                setCaption( value );
            }
        }
#endif

        public String getCaption() {
            return caption;
        }

        public void setCaption( String value ) {
            caption = value;
        }

#if !JAVA
        /// <summary>
        /// XMLシリアライズ用
        /// </summary>
        public int Length {
            get {
                return getLength();
            }
            set {
                setLength( value );
            }
        }
#endif

        public int getLength() {
            return length;
        }

        public void setLength( int value ) {
            length = value;
        }

        public String getDisplayString() {
            return IDS + caption;
        }

#if !JAVA
        public object Clone() {
            return clone();
        }
#endif

        public Object clone() {
            NoteHeadHandle result = new NoteHeadHandle();
            result.Index = Index;
            result.IconID = IconID;
            result.IDS = IDS;
            result.Original = Original;
            result.setCaption( getCaption() );
            result.setLength( getLength() );
            result.setDuration( getDuration() );
            result.setDepth( getDepth() );
            return result;
        }

        public VsqHandle castToVsqHandle() {
            VsqHandle ret = new VsqHandle();
            ret.m_type = VsqHandleType.NoteHeadHandle;
            ret.Index = Index;
            ret.IconID = IconID;
            ret.IDS = IDS;
            ret.Original = Original;
            ret.Caption = getCaption();
            ret.setLength( getLength() );
            ret.Duration = getDuration();
            ret.Depth = getDepth();
            return ret;
        }
    }

#if !JAVA
}
#endif
